medical-portal/src/API/RSBC.DMF.MedicalPortal.API/AutoMapper.cs

```diff
@@ -16,9 +25,97 @@ public MappingProfile()
                 .ForMember(dest => dest.FullName, opt => opt.MapFrom(src => src.Case.Person.FullName))
                 .ForMember(dest => dest.Birthday, opt => opt.MapFrom(src => src.Case.Person.Birthday))
                 .ForMember(dest => dest.ComplianceDate, opt => opt.MapFrom(src => src.ComplianceDate));
+
+
+            CreateMap<LegacyDocument, ViewModels.CaseDocument>()
+             .ForMember(dest => dest.SubmittalStatus, opt => opt.MapFrom(src => src.SubmittalStatus))
+             .ForMember(dest => dest.DueDate, opt => opt.MapFrom(src => DueDateConverter(src)))
+             .ForMember(dest => dest.CreatedOn, opt => opt.MapFrom(src => CreateDateConverter(src)))
+             .ForMember(dest => dest.DocumentType, opt => opt.MapFrom(src => src.DocumentType))
+             .ForMember(dest => dest.Description, opt => opt.MapFrom(src => src.Description));
+        }
+
+
+        private Expression<Func<string, string>> NullStringConverter = x => x ?? string.Empty;
+
+        private DateTimeOffset ImportDateConverter(LegacyDocument src)
```

Comment:
this is duplicate code that exists in driver portal. Duplicate code  should be refactored and moved to the Sharedutils project. In this case though, we can generalize this and probably remove this method altogether. What does this method to that this code doesn't:

  `CreateMap<Timestamp, DateTimeOffset>()
             .ConvertUsing(src => src.ToDateTimeOffset());`

Is converting the zero offset case important? I don't understand that part myself but it seems like this method is not needed. If it is needed, we should make a generic Timestamp to DateTimeOffset mapper by adjusting the code example above

Reply:
Agree I have removed the date conversion.